x-pack/elastic-agent/pkg/agent/transpiler/rules.go

```diff
@@ -175,6 +178,8 @@ func (r *RuleList) UnmarshalYAML(unmarshal func(interface{}) error) error {
 			r = &FixStreamRule{}
 		case "insert_defaults":
 			r = &InsertDefaultsRule{}
+		case "inject_custom_headers":
```

Comment:
Do we need this rule? Could the Agent code path just add it to the output correctly before passing it into the AST?

Reply:
this rule is to prevent from sending headers to processes  we dont want to share headers with.

Reply:
also discussed with @urso , that we will use capabilities for this later on.